components/brave_rewards/resources/page/components/pageWallet.tsx

```diff
@@ -557,7 +557,9 @@ class PageWallet extends React.Component<Props, State> {
     const totalBalance =
       !walletKeys.includes('blinded')
         ? undefined
-        : externalWallet && !walletKeys.includes(externalWallet.type)
+        : externalWallet
```

Comment:
This logic is getting pretty complex. You may want to move this into its own function where you can use early returns and other things to improve readability.

Reply:
Resolving as per our Slack conversation.